src/components/NewVault.tsx

```diff
@@ -10,24 +13,53 @@ const NewVault = ({ id }: Props) => {
     vaultMetrics,
     prices,
     priceErrors,
+    vaultFactoryParamsLoadingError,
   } = useVaultStore();
-
   const manager = vaultManagers.get(id);
   const metrics = vaultMetrics.get(id);
   const params = vaultGovernedParams.get(id);
   const brand = metrics?.retainedCollateral?.brand;
   const price = brand && prices.get(brand);
 
-  const error = vaultLoadingErrors.get(id) || (brand && priceErrors.get(brand));
+  const error =
+    vaultLoadingErrors.get(id) ||
+    (brand && priceErrors.get(brand)) ||
+    vaultFactoryParamsLoadingError;
+
+  const offerSigner = useAtomValue(offerSignerAtom);
+  const isReady =
+    manager &&
+    metrics &&
+    params &&
+    price &&
+    !error &&
+    offerSigner.isDappApproved;
+
+  const purses = useAtomValue(pursesAtom);
+  const istPurse = purses?.find(p => p.brand === metrics?.totalDebt?.brand);
+  const collateralPurse = purses?.find(
+    p => p.brand === metrics?.retainedCollateral?.brand,
+  );
+
+  const IST_TO_BORROW = 5_000_000n; // 5 IST
 
-  const isReady = manager && metrics && params && price && error === undefined;
+  // TODO: Calculate these based on user inputs and actual prices and liquidation ratios.
+  const proposeOffer = () => {
+    makeOpenVaultOffer(
+      collateralPurse?.pursePetname,
+      10_000_000_000n,
+      istPurse?.pursePetname,
+      IST_TO_BORROW,
+    );
+  };
 
   return (
     <button
-      className="bg-purple-500 hover:bg-purple-700 text-white font-semibold py-2 px-4 rounded"
+      className="disabled:bg-gray-300 disabled:cursor-not-allowed bg-purple-500 hover:bg-purple-700 text-white font-semibold py-2 px-4 rounded"
       disabled={!isReady}
+      onClick={() => proposeOffer()}
```

Comment:
It does work but that way passes the event arg to `proposeOffer`, which I thought would be cleaner to explicitly leave out. In the past with class components you had to be more careful though https://reactjs.org/docs/faq-functions.html

Reply:
IMO it's okay to send arguments to a function that ignores them but I agree here it's not necessary given how little performance impact it would have.